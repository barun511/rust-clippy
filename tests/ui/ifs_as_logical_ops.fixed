#![warn(clippy::ifs_as_logical_ops)]

fn main() {
    // test code goes here
}

fn john(b1: bool, b2: bool) -> bool {
    b2 && false
    //~^ ifs_as_logical_ops
}
