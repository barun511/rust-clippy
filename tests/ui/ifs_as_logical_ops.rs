#![warn(clippy::ifs_as_logical_ops)]

fn main() {
    // test code goes here
}

fn john(b1: bool, b2: bool) -> bool {
    if b1 { b2 } else { false }
    //~^ ifs_as_logical_ops
}
