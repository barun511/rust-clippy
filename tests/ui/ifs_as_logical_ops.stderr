error: Logical operations are clearer than if conditions in this instance
  --> tests/ui/ifs_as_logical_ops.rs:8:5
   |
LL |     if b1 { b2 } else { false }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `b2 && false`
   |
   = note: `-D clippy::ifs-as-logical-ops` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::ifs_as_logical_ops)]`

error: aborting due to 1 previous error

